### PR TITLE
PostGIS MVT provider and docs

### PR DESCRIPTION
**Pull request by PascalLike**
# Overview
This restores the possibility of using PostGIS+pg_tileserve as a tiles provider. This same provider works also for PostGIS+Martin without big changes, so there is no need to create a separate one. The provider is named after PostGIS, since it's the source for data.

# Related issue / discussion
https://github.com/geopython/pygeoapi/issues/1551

# Additional information
I set up a docker compose to run the tests on both pg_tileserver and Martin, if it helps the testing I can provide it.

# Dependency policy (RFC2)

- [X] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

# Updates to public demo

- [X] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [X] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [X] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: pygeoapi/plugin.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
         'MongoDB': 'pygeoapi.provider.mongo.MongoProvider',
         'MVT-tippecanoe': 'pygeoapi.provider.mvt_tippecanoe.MVTTippecanoeProvider',  # noqa: E501
         'MVT-elastic': 'pygeoapi.provider.mvt_elastic.MVTElasticProvider',  # noqa: E501
+        'MVT-postgis': 'pygeoapi.provider.mvt_postgis.MVTPostGISProvider',  # noqa: E501
```
Comment (francbartoli) [inline pygeoapi/plugin.py:55]: Can we change this name to a more generic `'MVT-proxy': 'pygeoapi.provider.mvt_proxy.MVTProxyProvider'` and ensure that in the docs the supported targets like `martin` and `pg_tileserv` are listed?

**Review by francbartoli — CHANGES_REQUESTED** [review]

File: pygeoapi/plugin.py (hunk 1 above)
Comment (PascalLike) [reply pygeoapi/plugin.py]: Thanks Francesco, I updated the code and the documentation following your suggestion.

**Review by francbartoli — APPROVED** [review]

File: docs/source/data-publishing/ogcapi-tiles.rst [hunk 2]
```diff
@@ -50,7 +51,7 @@ This code block shows how to configure pygeoapi to read Mapbox vector tiles gene
                  min: 0
                  max: 5
              schemes:
-                 - WorldCRS84Quad
```
Comment (doublebyte1) [inline docs/source/data-publishing/ogcapi-tiles.rst:53]: @PascalLike you can leave it like this, for now, but this parameter will disappear with my PR.

File: tests/pygeoapi-test-config-apirules.yml [hunk 3]
```diff
@@ -212,7 +212,7 @@ resources:
               data: tests/data/ne_110m_lakes.geojson
               id_field: id
             - type: tile
-              name: MVT
+              name: MVT-tippecanoe
```
Comment (doublebyte1) [inline tests/pygeoapi-test-config-apirules.yml:215]: well spotted!

**Comment by doublebyte1** [comment]
@PascalLike great work! Please add an example on [pygeoapi-examples](https://github.com/geopython/pygeoapi-examples), when you have the chance 👍🏽

**Review by doublebyte1 — APPROVED** [review]
Great job! Please add an example on [pygeoapi examples](https://github.com/geopython/pygeoapi-examples), when you have the chance

**Pull request merged**